root/etc/cont-init.d/02-tamper-check

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/with-contenv bash
+
+if ([ -d "/config/custom-cont-init.d" ] && [ -n "$(find /config/custom-cont-init.d ! -user root)" ]) || ([ -d "/config/custom-services.d" ] && [ -n "$(find /config/custom-services.d ! -user root)" ]); then
+    echo "**** Potential tampering with custom scripts/services detected ****"
+    randstr=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-8};echo;)
+    for folder in "/config/custom-cont-init.d" "/config/custom-services.d"; do
+        if [ -d "${folder}" ]; then
+            mv "${folder}" "${folder}.${randstr}"
+            echo "**** Folder ${folder} is moved to ${folder}.${randstr} ****"
+        fi
+    done
+    echo "**** The folders '/config/custom-cont-init.d' and '/config/custom-services.d'; and their contents need to all be owned by root to prevent root escalation inside the container!!! ****"
+    mkdir -p /config/custom-cont-init.d /config/custom-services.d
+    chown 0:0 /config/custom-cont-init.d /config/custom-services.d
```

Comment:
Spad suggested chmodding as well, otherwise looks good

Reply:
I think it's better not to chmod, but instead print a message in the log. As long as the folders are owned by root, it suggests that whoever tampered with the perms had root access. So the `other write` perms should only generate a warning without action to make sure that changed perms were intentionally done by an authorized user.